viewers/nerf_viewer/bridge_server/bridge_server_subprocess.py

```diff
@@ -0,0 +1,85 @@
+import atexit
+import os
+import signal
+import socket
+import subprocess
+import sys
+import threading
+import time
+
+import server
+
+from typing import Optional
+
+
+def is_port_open(port: int):
+    # check whether the given port is open
+    try:
+        sock = socket.socket()
+        sock.bind(("", port))
+        sock.close()
+
+        return True
+    except OSError:
+        return False
+
+
+def get_free_port(default_port: int = None):
+    if default_port:
+        if is_port_open(default_port):
+            return default_port
+    sock = socket.socket()
+    sock.bind(("", 0))
+    port = sock.getsockname()[1]
+
+    return port
+
+
+def run_bridge_server_as_subprocess(
+    websocket_port: int,
+    zmq_port: Optional[int] = None,
+    ip_address: str = "127.0.0.1",
```

Comment:
`0.0.0.0` may allow access to a larger range than `127.0.0.1`. Should we change the default value to `0.0.0.0`?

Reply:
Replace the default ip address to 0.0.0.0 makes zmq fail to connect with client.

`TODO: figure out why this happens.`